formats/src/main/scala/overflowdb/formats/graphml/GraphMLExporter.scala

```diff
@@ -0,0 +1,151 @@
+package overflowdb.formats.graphml
+
+import overflowdb.formats.{ExportResult, Exporter, isList}
+import overflowdb.{Element, Graph}
+
+import java.nio.file.{Files, Path}
+import scala.collection.mutable
+import scala.jdk.CollectionConverters.{IterableHasAsScala, IteratorHasAsScala, MapHasAsScala}
+
+/**
+ * Exports OverflowDB Graph to GraphML
+ *
+ * Note: GraphML doesn't natively support list property types, so we fake it by encoding it as a `;` delimited string.
+ * If you import this into a different database, you'll need to parse that separately.
+ * In comparison, Tinkerpop just bails out if you try to export a list property to graphml.
+ *
+ * https://en.wikipedia.org/wiki/GraphML
+ * http://graphml.graphdrawing.org/primer/graphml-primer.html
+ * */
+object GraphMLExporter extends Exporter {
+
+  override def runExport(graph: Graph, outputRootDirectory: Path) = {
+    val outFile = resolveOutputFile(outputRootDirectory)
+    val nodePropertyContextById = mutable.Map.empty[String, PropertyContext]
+    val edgePropertyContextById = mutable.Map.empty[String, PropertyContext]
+
+    val nodeEntries = graph.nodes().asScala.map { node =>
+      s"""<node id="${node.id}">
+         |    <data key="$KeyForNodeLabel">${node.label}</data>
+         |    ${dataEntries("node", node, nodePropertyContextById)}
+         |</node>
+         |""".stripMargin
+    }.toSeq
+
+    val edgeEntries = graph.edges().asScala.map { edge =>
+      s"""<edge source="${edge.outNode.id}" target="${edge.inNode.id}">
+         |    <data key="$KeyForEdgeLabel">${edge.label}</data>
+         |    ${dataEntries("edge", edge, edgePropertyContextById)}
+         |</edge>
+         |""".stripMargin
+    }.toSeq
+
+    val nodePropertyKeyEntries = nodePropertyContextById.map { case (key, PropertyContext(name, tpe, isList)) =>
+      val tpe0 =
+        if (isList) s"$tpe[]"
+        else tpe
+      s"""<key id="$key" for="node" attr.name="$name" attr.type="$tpe0"></key>"""
+    }.mkString("\n")
+    val edgePropertyKeyEntries = edgePropertyContextById.map { case (key, PropertyContext(name, tpe, isList)) =>
+      val tpe0 =
+        if (isList) s"$tpe[]"
+        else tpe
+      s"""<key id="$key" for="edge" attr.name="$name" attr.type="$tpe"></key>"""
+    }.mkString("\n")
```

Comment:
`System.lineSeparator` would be better in case someone runs it on Windows

Reply:
done